src/lib.rs

```diff
@@ -23,7 +23,7 @@ pub struct Clipboard {
 impl Clipboard {
     /// Creates new clipboard which will be running on its own thread with its own event queue to
     /// handle clipboard requests.
-    pub fn new(display: *mut c_void) -> Self {
+    pub unsafe fn new(display: *mut c_void) -> Self {
```

Comment:
Let's add a "Safety" section, telling that the pointer must be a valid `*mut wl_display` pointer, and that it must remain valid for as long as the `Clipboard` object is alive.